### PR TITLE
Added features to automatically award roles for donations and nitro boosting

### PR DESCRIPTION
**Pull request by skoleosho97**
This version provides two new features for the bot:

- Automatically reading through user donations, and assigning roles where applicable.
- Automatically checking if user uses Nitro on the server, or said Nitro expires, and assigns or removes roles where applicable.

New functions:

- `check_donation(message)`: Checks the given donation and determines whether the user is in the server.
- `donation(target, amount)`: Analyzes donation amount and assigns the user any applicable roles.
- `is_boosted(user)`: Checks if the user has boosted the server.

I am unsure whether or not the user can have multiple `Custom //` roles, so I wrote the functions assuming that you can. If you cannot, I can adjust them right away.

### CODE REVIEW

**Comment by jotslo** [comment]
Apologies for the extremely late response, we are going to be making modifications that make contributing much easier, and adding new contributors to the repository shortly to ensure that the bot continues to stay maintained.

Thank you for your contribution, unfortunately we can't accept your pill request just yet because the implementation of custom colours isn't entirely correct. A user can only have one custom colour at a time. Additionally, the user is immediately awarded the Gold custom colour when they donate. This needs to be changed to check whether they already have an existing custom colour beforehand.

These will need to be resolved before the pull request can be merged.

**Comment by skoleosho97** [comment]
Will have this updated by the end of the day, thanks for the response!

**Comment by skoleosho97** [comment]
Added fix!

File: Modules/data.py [hunk 1]
```diff
@@ -12,19 +12,24 @@
...
+#donator = 323047035840233474
+#donator_plus = 325937099557830656
+#custom_gold = 639147821273972736
+
 no_server = 676921838181548083
 no_gamejam = 678974586951565312
 no_event = 676921944880185365
 
-general_channel = 306153640023031820
+general_channel = 762530852977901622 # -- replace with actual general_channel
```
Comment (spaceraiders0) [inline Modules/data.py]: Was this a mistake? The actual channel id for the server is 306153640023031820.

File: Modules/data.py [hunk 2]
```diff
@@ -37,7 +42,7 @@
     462394160629153812: 2,
     699995169629667370: 2,
     379023623245004810: 2,
-    306885827189800960: 3,
+    762530852977901622: 3, # -- replace with actual general_channel
```
Comment (spaceraiders0) [inline Modules/data.py]: Same question as [here](https://github.com/ScriptersCF/mr-script-bot/pull/25/files#r697895643).

**Comment by spaceraiders0** [comment]
I like what you have got here so far. Aside from the merge conflicts, which I will deal with most likely myself, it would be great if you could resolve the comments that I have posted. The main problem I noticed is simply with the IDs being incorrect, which I imagine would have been because you are testing the bot on a private server. I will put comments on each occurrence I can find of this.

Aside from that, it would be great if you could potentially add docstrings to your functions, as it makes code seem a bit cleaner, and explains the intent more than single-line comments. Any form is fine, but I can suggest something like the [Sphinx format](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html).

If you have any questions let me know.

File: Modules/data.py [hunk 3]
```diff
@@ -12,19 +12,24 @@
 aban = 550114698297737216
 shban = 416786636118949918
 
-verified = 346604542550605835
+verified = 762535941579341838
```
Comment (spaceraiders0) [inline Modules/data.py]: Correct Id should be 346604542550605835.

**Comment by skoleosho97** [comment]
Thanks for getting back to me, I'll address these concerns right away!

File: Modules/data.py (hunk 3 above)
Comment (skoleosho97) [reply Modules/data.py]: Made some changes just now, let me know if anything else needs to be addressed!

File: Modules/data.py [hunk 4]
```diff
@@ -42,7 +42,7 @@
     462394160629153812: 2,
     699995169629667370: 2,
     379023623245004810: 2,
-    762530852977901622: 3, # -- replace with actual general_channel
+    306885827189800960: 3,
```
Comment (spaceraiders0) [inline Modules/data.py]: Might have been a mistake on my part, but this is not the correct id fro the general channel.
The correct id is: ``306885827189800960``.

**Comment by spaceraiders0** [comment]
Thank you for your contribution!

**Pull request merged**